helmfile/values/networkpolicy/service/grafana.yaml.gotmpl

```diff
@@ -0,0 +1,85 @@
+policies:
+  monitoring:
+    grafana:
+      podSelectorLabels:
+        app.kubernetes.io/name: grafana
+      egress:
+        - rule: egress-rule-dns
+        - rule: egress-rule-apiserver
+        - name: egress-rule-prometheus
+          peers:
+            - podSelectorLabels:
+                app.kubernetes.io/name: prometheus
+            - podSelectorLabels:
+                app: grafana-label-enforcer
+          ports:
+            - tcp: 9090
+        - name: egress-rule-thanos
+          peers:
+            - namespaceSelectorLabels:
+                kubernetes.io/metadata.name: thanos
+              podSelectorLabels:
+                app.kubernetes.io/component: query-frontend
+          ports:
+            - tcp: 10902
+        - name: egress-rule-dex
+          peers:
+            - namespaceSelectorLabels:
+                kubernetes.io/metadata.name: dex
+              podSelectorLabels:
+                app.kubernetes.io/name: dex
+          ports:
+            - tcp: 5556
+        {{- with .Values.networkPolicies.monitoring.grafana.externalDataSources }}
+        {{- if .enabled }}
+        - name: egress-rule-external-datasources
+          {{- with .ips }}
+          peers:
+            {{- range . }}
+            - cidr: {{ . }}
+            {{- end }}
+          {{- end }}
+          {{- with .ports }}
+          ports:
+            {{- range . }}
+            - tcp: {{ . }}
+            {{- end }}
+          {{- end }}
+        {{- end }}
+        {{- end }}
+        {{- with .Values.networkPolicies.monitoring.grafana.externalDashboardProvider }}
+        - name: egress-rule-external-dashboards
+          {{- with .ips }}
+          peers:
+            {{- range . }}
+            - cidr: {{ . }}
+            {{- end }}
+          {{- end }}
+          {{- with .ports }}
+          ports:
+            {{- range . }}
+            - tcp: {{ . }}
+            {{- end }}
+          {{- end }}
+        {{- end }}
```

Comment:
This doesn't look that nice, but I imagine that when we fully use this method we can change the apps config to use the same format as the chart and add validation that IPs and ports are set.

Reply:
Somewhat simplified now.